modules/core/src/main/java/org/apache/ignite/internal/processors/cache/IgniteCacheProxyImpl.java

```diff
@@ -1288,9 +1293,26 @@ private void validate(Query qry) {
         }
     }
 
+    /**
+     * Warns if an ineffective annotation is used
+     *
+     * @param valueCls Value class
+     */
+    private void checkIneffectiveAnnotations(Class valueCls) {
+        for (Field f:valueCls.getDeclaredFields()) {
+            AffinityKeyMapped[] annotations = f.getDeclaredAnnotationsByType(AffinityKeyMapped.class);
+            if (annotations.length > 0) {
+                LT.warn(log,
+                        "AffinityKeyMapped annotation should be used on key and not value type");
+            }
+
+        }
+    }
+
     /** {@inheritDoc} */
     @Override public void put(K key, V val) {
         IgniteInternalCache<K, V> delegate = getDelegateSafe();
+        checkIneffectiveAnnotations(val.getClass());
```

Comment:
Reflection use on each put, really? it will kill performance. Please move this check to new type registration method, I believe you will find it in BinaryMarshaller or somewhere around.

Reply:
Thank you, I called out the performance cost of reflection on the [issue](https://issues.apache.org/jira/browse/IGNITE-12702?jql=project%20%3D%20IGNITE%20AND%20labels%20in%20(newbie)%20and%20status%20%3D%20OPEN) itself almost a month back. Didn't know about BinaryMarshaller, will use it

Reply:
@Mikhail1988 Where can I find new type registration method? I did see binary type configurator